Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -858,6 +858,8 @@ void BlockFormattingContext::layout_block_level_box(Box const& box, BlockContain
             m_margin_state.reset();
         }
         m_y_offset_of_current_block_container = box_state.offset.y() + box_state.content_height() + box_state.border_box_bottom();
+    } else if (creates_block_formatting_context(block_container) && block_container.is_absolutely_positioned()) {
+        m_y_offset_of_current_block_container = box_state.offset.y() + box_state.content_height() + box_state.border_box_bottom();
```

Comment:
All tests pass for me as well when I add ` || block_container.is_out_of_flow()` to the existing `if` condition above. Are you sure the margin state should not be reset?

Either I missed something that your test does not include, or this could be a bit simpler :)